### DISCUSSION TITLE
[help] How can I force to show output at top of the terminal?

### DISCUSSION THREAD

**Discussion by avatar-lavventura** [dandavison/delta#1417] [General]
When I pipe into `delta` it shows the output bottom of the terminal instead of showing it top of the page.

Example output:

<img width="589" alt="My Screenshot 2023-05-16 at 15 59 27" src="https://github.com/dandavison/delta/assets/18537398/70a070ec-db38-4ae5-99dd-c07071235d5b">

When I press to `n` it is corrected as:
<img width="595" alt="My Screenshot 2023-05-16 at 15 59 55" src="https://github.com/dandavison/delta/assets/18537398/d2d08bdb-849e-439d-8cab-895b2a4321fd">

By defualt is it possible to force `delta` to show its output at top of the terminal?

**Comment by dandavison** [comment]
Hi there, it's actually the pager `less` that is responsible for this behaviour, not delta. Here's what you need to do:


1. Work out the exact command-line flags that you want to pass to less to get the behaviour you want. For example, you could use a command like this:
```
yes | head -n 2 | LESS= less <your command line flags here>
```

2. Make sure that delta is using those flags when invoking `less`. Unfortunately there are various ways of doing this. Let's suppose you want `-FRS`. you can store them in the `LESS` env var: `export LESS=-FRS` or you can set an env var: `export DELTA_PAGER='less -FRS`, or you can even use the `delta --pager='less -FRS'` command or use `pager = less -FRS` in the `[delta]` section of your gitconfig.

Sorry there are so many ways; I promise there is a reason for each one!

**Comment by avatar-lavventura** [comment]
No need to sorry :) I have tried: `yes | head -n 2 | delta --pager='less -r +A'` and it worked like charm.

**Comment by scanny** [comment]
I found that `delta --pager='less -c'` did the trick for me.

In `less` parlance of "screen repaints", `-c` or `--clear-screen` "Causes full screen repaints to be painted from the top line down. By default, full screen repaints are done by scrolling from the bottom of the screen."

Using the alternate buffer, as I do, is a "full-screen repaint".

I prefer this behavior for git diffs because my eyes are used to scanning from the top of the screen on diffs, so always having them start there prevents a "flow-interruption" when my eyes automatically look there, don't find anything, and have to scan downward.
